html/webappapis/dynamic-markup-insertion/opening-the-input-stream/beforeunload.window.js

```diff
@@ -0,0 +1,12 @@
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  t.add_cleanup(() => frame.remove());
+  frame.src = "/common/blank.html";
+  frame.onload = t.step_func(() => {
+    frame.contentWindow.onbeforeunload = t.unreached_func("beforeunload should not be fired");
+    frame.contentDocument.open();
+    t.step_timeout(t.step_func_done(() => {
+      frame.contentDocument.close();
```

Comment:
Can you add a comment describing why it's helpful to invoke `document.close()` here? I don't quite see why that would fire the beforeunload event, which is the only thing that would change the outcome of the test...

Reply:
Sure. The intent here is to test that `open()` does _not_ fire the `beforeunload` event. The `close()` here is merely to fire the `load` event on the document and unblock testharness from thinking that the page has not yet loaded.

Reply:
Added a comment. LMK if that works.